forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/listeners/TaskAssignmentListener.java

```diff
@@ -16,77 +25,180 @@
-        String taskId = String.valueOf(delegateTask.getId());
-
-        // Create empty request and interceptors for the SendMailInvocation
-        SendMailRequest request = getRequest();
-        List<ConnectorRequestInterceptor> requestInterceptors = getInterceptors();
-
-        if (recipient != null && !recipient.isEmpty()) {
+        try {
+            // Get the string value of the variables passed from the task
+            String emailBody = getBody(delegateTask.getExecution());
+            String emailSubject = getSubject(delegateTask.getExecution());
+            String taskId = String.valueOf(delegateTask.getId());
+            Boolean attachPdf = getAttachSubmission(delegateTask.getExecution());
+            InternetAddress[] recipients = getRecipients(delegateTask.getExecution());
+
+            // Create empty request and interceptors for the SendMailInvocation
+            SendMailRequest request = getRequest();
+            List<ConnectorRequestInterceptor> requestInterceptors = getInterceptors();
+
+            if (attachPdf) {
+                // Get the form fields and values data to use in the pdf template
+                Map<String, Object> data = getFormPdfData(delegateTask.getExecution());
+
+                // Generate the template from '/templates/form.html'
+                Template template = getTemplate();
+
+                // Create a temporary 'output.html' file template
+                File f = null;
+                try {
+                    f = File.createTempFile("output", ".html");
+                    template.process(data, new FileWriter(f));
+
+                    // Render the file as 'form.pdf'
+                    RenderPage(f);
+                } finally {
+                    if (f != null) {
+                        f.deleteOnExit();
+                    }
+                }
+            }
 
-            try {
-                Message message = createMessage(recipient, emailBody, emailSubject, taskId, mailService.getSession());
-                SendMailInvocation invocation = new SendMailInvocation(message, request, requestInterceptors, mailService);
+            if (recipients.length > 0) {
+                try {
+                    Message message = createMessage(recipients, emailBody, emailSubject, taskId, attachPdf, mailService.getSession());
+                    SendMailInvocation invocation = new SendMailInvocation(message, request, requestInterceptors, mailService);
 
-                invocation.proceed();
+                    invocation.proceed();
 
-            } catch (Exception e) {
-                throw new MailConnectorException("Failed to send mail: " + e.getMessage(), e);
+                } catch (Exception e) {
+                    throw new MailConnectorException("Failed to send mail: " + e.getMessage(), e);
+                }
             }
-
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
I think we should try to handle this exception gracefully instead of just printing the exception.

A pattern used in other Listeners (for example FormBPMDataPipelineListener) would be to move all this logic into a separate method, and passing any exception along to `handleException` of the `BaseListener`.

Example:

```java
    @Override
    public void notify(DelegateTask delegateTask) {
        try {
            sendEmail(delegateTask.getExecution());
        } catch (Exception e) {
            handleException(delegateTask.getExecution(), ExceptionSource.TASK, e);
        }
    }
```